File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/javadoc/comment/direct/Java/Go/s573993837_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

var (
	L, R, top, bottom int
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	solve(reader, writer)
}

func solve(reader *bufio.Reader, writer *bufio.Writer) {
	t := 1
	for t > 0 {
		line, _ := reader.ReadString('\n')
		inputs := strings.Fields(line)
		n, _ := strconv.ParseInt(inputs[0], 10, 64)
		m, _ := strconv.ParseInt(inputs[1], 10, 64)

		if n == 0 {
			fmt.Fprintln(writer, 0)
			t--
			continue
		}
		if n == m {
			fmt.Fprintln(writer, 0)
			t--
			continue
		}
		if n > m {
			n = m + n%m
			n -= m
			fmt.Fprintln(writer, min(m-n, n))
			t--
			continue
		}
		fmt.Fprintln(writer, min(m-n, n))
		t--
	}
}

func min(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

func swap(chrr []rune, i, j int) {
	chrr[i], chrr[j] = chrr[j], chrr[i]
}

func num(n int) int {
	a := 0
	for n > 0 {
		a += n & 1
		n >>= 1
	}
	return a
}

func isPrime(n int) bool {
	if n <= 1 {
		return false
	}
	if n <= 3 {
		return true
	}
	if n%2 == 0 || n%3 == 0 {
		return false
	}
	for i := 5; i*i <= n; i += 6 {
		if n%i == 0 || n%(i+2) == 0 {
			return false
		}
	}
	return true
}

func gcd(a, b int64) int64 {
	if b == 0 {
		return a
	}
	if a > b {
		return gcd(b, a%b)
	}
	return gcd(a, b%a)
}

func fastPow(base, n, M int64) int64 {
	if n == 0 {
		return 1
	}
	if n == 1 {
		return base
	}
	halfN := fastPow(base, n/2, M)
	if n%2 == 0 {
		return (halfN * halfN) % M
	}
	return (((halfN * halfN) % M) * base) % M
}

func modInverse(n, M int64) int64 {
	return fastPow(n, M-2, M)
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/C++/s573993837_commented.cpp
#include <iostream>
#include <string>
#include <queue>
#include <sstream>
#include <vector>

using namespace std;

static int L, R, top, bottom;

class InputReader {
private:
    istream& in;
    queue<string> tokens;
public:
    InputReader(istream& is) : in(is) {}
    string next() {
        if (tokens.empty()) {
            string line;
            getline(in, line);
            istringstream iss(line);
            string token;
            while (iss >> token) {
                tokens.push(token);
            }
        }
        if (tokens.empty()) return "";
        string result = tokens.front();
        tokens.pop();
        return result;
    }
    int nextInt() { return stoi(next()); }
    long long nextLong() { return stoll(next()); }
    double nextDouble() { return stod(next()); }
};

void solve(InputReader& sc, ostream& out) {
    int t = 1;
    u_loop:
    while (t-- > 0) {
        long long n = sc.nextLong();
        long long m = sc.nextLong();
        if (n == 0) {
            out << "0\n";
            continue u_loop;
        }
        if (n == m) {
            out << "0\n";
            continue u_loop;
        }
        if (n > m) {
            n = m + n % m;
            n -= m;
            out << min(m - n, n) << '\n';
            continue u_loop;
        }
        out << min(m - n, n) << '\n';
    }
}

void swap(char arr[], int i, int j) {
    char temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
}

int num(int n) {
    int a = 0;
    while (n > 0) {
        a += (n & 1);
        n >>= 1;
    }
    return a;
}

bool isPrime(int n) {
    if (n <= 1) return false;
    if (n <= 3) return true;
    if (n % 2 == 0 || n % 3 == 0) return false;
    for (int i = 5; i * i <= n; i += 6) {
        if (n % i == 0 || n % (i + 2) == 0)
            return false;
    }
    return true;
}

long long gcd(long long a, long long b) {
    if (b == 0) return a;
    return (a > b) ? gcd(b, a % b) : gcd(a, b % a);
}

long long fast_pow(long long base, long long n, long long M) {
    if (n == 0) return 1LL;
    if (n == 1) return base % M;
    long long halfn = fast_pow(base, n / 2, M);
    if (n % 2 == 0)
        return (halfn * halfn) % M;
    else
        return ((halfn * halfn) % M * base) % M;
}

long long modInverse(long long n, long long M) {
    return fast_pow(n, M - 2, M);
}

void feedArr(long arr[], int size, InputReader& sc) {
    for (int i = 0; i < size; ++i) {
        arr[i] = sc.nextLong();
    }
}

void feedArr(int arr[], int size, InputReader& sc) {
    for (int i = 0; i < size; ++i) {
        arr[i] = sc.nextInt();
    }
}

void feedArr(double arr[], int size, InputReader& sc) {
    for (int i = 0; i < size; ++i) {
        arr[i] = sc.nextDouble();
    }
}

void feedArr(string arr[], int size, InputReader& sc) {
    for (int i = 0; i < size; ++i) {
        arr[i] = sc.next();
    }
}

string printArr(int arr[], int size) {
    ostringstream oss;
    for (int i = 0; i < size; ++i) {
        oss << arr[i] << " ";
    }
    return oss.str();
}

string printArr(long arr[], int size) {
    ostringstream oss;
    for (int i = 0; i < size; ++i) {
        oss << arr[i] << " ";
    }
    return oss.str();
}

string printArr(string arr[], int size) {
    ostringstream oss;
    for (int i = 0; i < size; ++i) {
        oss << arr[i] << " ";
    }
    return oss.str();
}

string printArr(double arr[], int size) {
    ostringstream oss;
    for (int i = 0; i < size; ++i) {
        oss << arr[i] << " ";
    }
    return oss.str();
}

struct Pair {
    int u, v, i, val;
    Pair(int a, int b, int i) : u(a), v(b), i(i) {}
};

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    InputReader in(cin);
    ostream& out = cout;
    solve(in, out);
    return 0;
}
// End of Code
